pkg/apply/applier.go

```diff
@@ -170,8 +170,14 @@ func (a *Applier) prepareObjects(infos []*resource.Info) (*ResourceObjects, erro
 		}
 	}
 
+	// Get the resources in the specified namespace from inventory object
+	resources = getNamespacedResources(resources, invs[0].Namespace)
+	if len(resources) == 0 {
```

Comment:
It is not clear to me what we are doing here. The only thing we want to check here is that the namespace used in the inventory object isn't something we are trying to apply. In fact, the key here is that the namespace must already exist.
I haven't thought about this before, but we might also want to verify that we don't end up pruning the namespace where the inventory object is stored. This is a bit of a strange case though

Reply:
I think I add this because one of the test saying, it should fail if there is only inventory object without resources.

Reply:
So it seems like in the `getNamespacedResources` function, we are filtering out any resources that are either cluster-scoped or namespace-scoped, but in a different namespace than the inventory. We don't want to do this. We always want all the resources that are being applied to be referenced in the inventory object.
We only care about cluster-scoped resources and the namespace of all the different resources in the `init` command when we try to determine the right namespace to use in the inventory template.

If someone is trying to apply a set of resources that only includes the inventory template and no other resources, I agree that should probably be considered an error.

Reply:
getNamespacedResources drop the resources that are :
 1. namespace scoped but in a different namespace than the inventory object, which was set or decided in the init command
2. non-builtin cluster scoped resources

Reply:
chatted offline on this, reverted changes of dropping namespaced resources that is in the same namespace as inventory object.